proteus/mprans/BoundaryConditions.py

```diff
@@ -331,117 +455,193 @@ def setTwoPhaseVelocityInlet(self, U, waterLevel, vert_axis=None, air=1.,
-        """Imposes a hydrostatic pressure profile and open boundary conditions
-        with a known otuflow depth
-       :param rhoUp: Phase density of the upper part.
-       :param rhoDown: Phase density of the lower part.
-       :param g: Gravitational acceleration vector.
-       :param refLevel: Level at which pressure = pRef.
-       :param pRef: Reference value for the pressure at x[vert_axis]=refLevel,
-                  be default set to 0.
-       :param vert_axis: index of vertical in position vector, must always be
-                       aligned with gravity, by default set to 1.
-       :return: hydrostaticPressureOutlet except when the pressure and the
-                vof are defined. Then it returns the pressure and vof profile
-                based on the known depth.
-       If the boundary is aligned with one of the main axes, sets the
-       tangential velocity components to zero as well.
-       (!) This condition is best used for boundaries and gravity aligned with
-           one of the main axes.
-       """
-        # self.reset()
-
-        # if vert_axis is None:
-        #     vert_axis = self.Shape.Domain.nd - 1
-
-        # def hydrostaticPressureOutletWithDepth_p_dirichlet(x, t):
-        #     if x[vert_axis] < seaLevel:
-        #         a0 = pRef-rhoUp*g[vert_axis]*(refLevel-seaLevel)-rhoDown*g[vert_axis]*seaLevel
-        #         a1 = rhoDown*g[vert_axis]
-        #         return a0 + a1*x[vert_axis]
-
-        # def hydrostaticPressureOutletWithDepth_vof_dirichlet(x, t):
-        #     if x[vert_axis] < seaLevel:
-        #         return water
-
-        # self.setHydrostaticPressureOutlet(rhoUp, g, refLevel, pRef, vert_axis)
-        # self.p_dirichlet.uOfXT = hydrostaticPressureOutletWithDepth_p_dirichlet
-        # self.vof_dirichlet.uOfXT = hydrostaticPressureOutletWithDepth_vof_dirichlet
-        pass
+        """
+        Returns the pressure and vof profile based on the known depth.
+        If the boundary is aligned with one of the main axes, sets the tangential
+        velocity components to zero as well.
+        (!) This condition is best used for boundaries and gravity aligned with
+            one of the main axes.
+        
+        Parameters
+        ----------        
+        rhoUp: Phase density of the upper part.
+        rhoDown: Phase density of the lower part.
+        g: Gravitational acceleration vector.
+        refLevel: Level at which pressure = pRef.
+        pRef: Reference value for the pressure at x[vert_axis]=refLevel, by default set to 0.
+        vert_axis: index of vertical in position vector, must always be aligned with gravity, by default set to 1.
+        """
+        self.reset()
 
+        if vert_axis is None:
+            vert_axis = self.nd-1
+
+        def hydrostaticPressureOutletWithDepth_p_dirichlet(x, t):
+            if x[vert_axis] < seaLevel:
+                a0 = pRef-rhoUp*g[vert_axis]*(refLevel-seaLevel)-rhoDown*g[vert_axis]*seaLevel
```

Comment:
I've used a derivation of the hydrostatic pressure that integrates it exactly over the two phases based on the smoothed Heaviside that we use to weight the densities. It may not really matter, and I'll look at that tweak later.